File: Sales_P3/script/masterdata.py
import pandas as pd


def master_cc():
    return pd.read_csv("meta/CC_2023.csv", dtype="str")


def master_gl():
    return pd.read_csv("meta/GL.csv", dtype="str")


def master_ph():
    # Read the PH info CSV file and rename columns
    df = pd.read_csv("meta/PH info.csv", dtype="str")
    df = df.rename(
        columns={
            "Profit Center": "Profit Ctr",
            "Product Hierarchy": "Product Hierarchy",
            "PH_3 simple": "PH_3 simple",
            "PRD/MER": "PRD/MER",
        }
    )

    # Select relevant columns and return the result
    return df[["Profit Ctr", "Product Hierarchy", "PH_3 simple", "PRD/MER"]]


def master_poc():
    return pd.read_csv("../PnL/meta/POC.csv", dtype="str")


def master_cm_cluster():
    return pd.read_excel("meta/YPC1 costing_Icheon.xlsx", engine="openpyxl")


def customer_material():
    # Read Excel file
    df = pd.read_excel(
        "meta/Customer Material.xlsx",
        sheet_name="Sheet1",
        engine="openpyxl",
        dtype=str,
    )

    # Select relevant columns and drop rows with missing values in Customer Material
    df = df[["Product", "Customer Material"]].dropna(
        subset=["Customer Material"])

    return df


mat_0180 = "meta/Material master_0180.xlsx"
mat_2182 = "meta/Material master_2182.xlsx"


def master_material(path):
    # read excel file
    df = pd.read_excel(path, sheet_name="MM", engine="openpyxl", dtype=object)

    # rename columns
    df = df.rename(
        columns={
            "Product Hierachy": "Product Hierarchy",
            "Material": "Product",
            "Profit Center": "Profit Ctr",
        }
    )

    # select relevant columns
    df = df[
        [
            "Product",
            "Profit Ctr",
            "Material type",
            "Product Hierarchy",
            "Standard Price",
            "Ext. Matl. Group",
        ]
    ]

    return df


def master_mat_0180():
    return master_material(mat_0180)


def master_mat_2182():
    return master_material(mat_2182)


def material_master():
    df_0180 = master_mat_0180()
    df_2182 = master_mat_2182()
    return pd.concat([df_0180, df_2182], ignore_index=True)

File: Sales_P3/src/dplyr/master_data.R
library(dplyr)
library(readr)
library(readxl)


# Master data ----
master_cc <- function() {
  df <- read_csv("meta/CC_2023.csv", show_col_types = FALSE)
  return(df)
}

master_gl <- function() {
  df <- read_csv("meta/GL.csv", show_col_types = FALSE)
  return(df)
}

master_ph <- function() {
  df <- read_csv("meta/PH info.csv", show_col_types = FALSE) |>
    rename(c(
      `Profit Ctr` = "Profit Center"
    )) |>
    select(c(
      "Profit Ctr",
      "Product Hierarchy",
      "PH_3 simple",
      "PRD/MER"
    ))
  return(df)
}

master_poc <- function() {
  df <- read_csv("../PnL/meta/POC.csv", show_col_types = FALSE)
  return(df)
}

master_cm_cluster <- function() {
  df <- read_excel("meta/YPC1 costing_Icheon.xlsx")
  return(df)
}

customer_material <- function() {
  df <- read_excel("meta/Customer Material.xlsx", sheet = "Sheet1") |>
    select(c(
      "Product",
      "Customer Material"
    )) |>
    filter(df["Customer Material"] != "NA")
  return(df)
}


# Material master ----
mat_0180 <- "meta/Material master_0180.xlsx"
mat_2182 <- "meta/Material master_2182.xlsx"

master_material <- function(path) {
  df <- read_excel(path, sheet = "MM")

  df <- df |>
    rename(c(
      `Product Hierarchy` = "Product Hierachy...8"
    )) |>
    select(c(
      "Material",
      "Profit Center",
      "Material type",
      "Product Hierarchy",
      "Standard Price",
      "Ext. Matl. Group"
    )) |>
    rename(c(
      Product = "Material",
      `Profit Ctr` = "Profit Center",
    ))

  return(df)
}

master_mat_0180 <- function() {
  df <- master_material(mat_0180)
  return(df)
}

master_mat_2182 <- function() {
  df <- master_material(mat_2182)
  return(df)
}

material_master <- function() {
  df <- bind_rows(master_mat_0180(), master_mat_2182())
  return(df)
}
